cmake/app/boilerplate.cmake

```diff
@@ -256,8 +256,22 @@ set(CMAKE_CXX_COMPILER_FORCED 1)
 
 include(${ZEPHYR_BASE}/cmake/version.cmake)
 include(${ZEPHYR_BASE}/cmake/host-tools.cmake)
+
+# DTS should be close to kconfig because CONFIG_ variables from
+# kconfig and dts should be available at the same time.
+#
+# The DT system uses a C preprocessor for it's code generation needs.
```

Comment:
@galak Does DTS actually require the target-toolchain to run? Would plain gcc not be sufficient if it only requires the preprocessor?

If so, then we could avoid O(n)-many `generic.cmake` files.

Reply:
It doesn't require the target-toolchain. It requires 'any' C toolchain (preprocessor).

'gcc' is not sufficient. Because 'gcc' is not a dependency, so we can't avoid O(n)-many generic.cmake files.

Said differently, Zephyr does not require a host C toolchain to be installed, but can rely solely on a single cross-compiling toolchain.

Reply:
IMHO, since the C toolchain is just used as a code generator, we should long-term migrate DT to use Jinja instead of CPP, and merge back together the toolchain config files. But this will have to wait for now ...

Reply:
Yes.

Technically #include is not needed because DT has include, but it does have it.

http://jinja.pocoo.org/docs/2.10/templates/#include
http://jinja.pocoo.org/docs/2.10/templates/#assignments
http://jinja.pocoo.org/docs/2.10/templates/#macros

Reply:
I see, will I don't think we will move to using jinja for DTS files as we want to share DTS with Linux Kernel or other systems.

Reply:
@SebastianBoe True, but ZephyrSDK is a dependency and there we pack a couple of gcc's. We could pick a preprocessor from there?

Reply:
>  ZephyrSDK

Not for all platforms.